bin/generate_charts.py

```diff
@@ -188,177 +186,168 @@ def create_chart(prefix, workload, title, y_label, time_series):
             max_y = y
     chart.range = (max_y * 0.0, max_y * 1.20)
 
-    file_list[prefix].append(f'{workload}.svg')
-    chart.render_to_file(f'{output}{workload}.svg')
+    return chart.render(disable_xml_declaration=True)
 
 
 def generate_charts(files):
-
-    aggregate = []
+    workloads = {}
 
     # Charts are labeled based on benchmark names, we need them
     # to be unique. A combination of (driver, workload) defines
     # a unique benchmark and name is "{driver}-{workload}".
     benchmark_names = set()
     for file in sorted(files):
         data = json.load(open(file))
+        data['workload'] = data['workload'].replace('/', '-')
         begin_time = data['beginTime'] if 'beginTime' in data.keys() else ""
         rp_version = data['version'] if 'version' in data.keys() and data['version'] else "unknown_version"
         unique_name = "{ver}-{ts}-{driver}-{workload}".format(ver=rp_version, ts=begin_time, driver=data['driver'], workload=data['workload'])
         # name used as chart label.
         name = "{ver}-{driver}-{workload}".format(ver=rp_version, driver=data['driver'], workload=data['workload'])
         if unique_name in benchmark_names:
-            print(f"Duplicate benchmark found: {name} in file {file}")
-            exit(-1)
+            print(f"WARN: Duplicate benchmark found: {name} in file {file}", file=sys.stderr)
+
         benchmark_names.add(unique_name)
+
+        if coalesce:
```

Comment:
@tmgstevens Curious how this looks, you have an example that you can share? (privately is ok too).

Reply:
So this defaults to false which is existing behaviour - so in the default load, each workload will output a JSON object of aggregated stats. With coalesce turned on, all workloads will be put into the same JSON object. I'm hoping your existing tooling wouldn't need to turn that on.

The only risk here is that where we'd previously broken the workload identification code, you may be expecting more JSON objects dumped out now??

Reply:
Or fewer - you may need to turn coalesce on actually - which wouldn't be a massive change. I think we've now captured the original (not coalesced) and new behaviour (coalesced)